### ISSUE TITLE
Invalid memory access in LOR mesh constructor in 2D

### ISSUE THREAD

**Issue by barker29** [mfem/mfem#129]
Vertex.SetCoords (vertex.hpp line 44) reads 3 doubles regardless of dimension, so when it is called from the LOR mesh constructor in mesh.cpp around line 2928:

```
vertices[rdofs[i]].SetCoords(phys_pts.GetColumn(i));
```

it reads from invalid memory in 2D, because phys_pts was initialized as a 2 by 2 matrix a few lines earlier.

This is normally not a problem because the read data is not used for anything, but in principle this could probably cause a segfault. To recreate the problem run the following code snippet through valgrind:

```
#include "mfem.hpp"

using namespace mfem;

int main(int argc, char *argv[])
{
   const char *mesh_file = "../data/star.mesh"; // invalid read from Valgrind
   // const char *mesh_file = "../data/fichera.mesh"; // no Valgrind errors

   int order = 3;
   int basis = BasisType::GetType('G');

   Mesh *mesh = new Mesh(mesh_file, 1, 1);
   Mesh *mesh_lor = new Mesh(mesh, order, basis);

   delete mesh;
   delete mesh_lor;

   return 0;
}
```

**Comment by v-dobrev** [comment]
Good catch. The easiest solution, I see, is to add an `int dim` parameter to `SetCoords()` and update all calls with the correct dimension, i.e. the space-dimension.

**Comment by tzanio** [comment]
Merged in 806cf543c047f69feb7f1398d578f914ffdccf9b
